### DISCUSSION TITLE
How to get IStorageItem (or IStorageBookmarkItem) for Android app-specific files?

### DISCUSSION THREAD

**Discussion by segln** [AvaloniaUI/Avalonia#14455] [Q&A] [answered]
[App-specific files](https://developer.android.com/training/data-storage/app-specific)

In Android, you can access app-specific files without permission in `/data/user/0/appname/` or `/storage/emulated/0/Android/data/appname/`.
And you can use .NET APIs like `System.IO.File.WriteAllText` or `System.IO.Directory.CreateDirectory` to access.

But how do I obtain an `IStorageItem` object that points to that path?
I tried `StorageProvider` methods but they asked for READ_EXTERNAL_STORAGE permission, or didn't work.

\+ Is there property or method returns `/data/user/0/appname/` directly in Avalonia? I'm using `Environment.GetFolderPath` or `AppDomain.CurrentDomain.BaseDirectory` currently, is this recommended way?

**Comment by maxkatz6** [comment, marked as answer]
Yes, using Environment.GetFolderPath is perfectly valid, when platform supports it. You wouldn't be able to do that in Browser though.

But I would expect StorageProvider to also work. Currently these methods always require READ_EXTERNAL_STORAGE permission, even if file path isn't external. This should be fixed https://github.com/AvaloniaUI/Avalonia/blob/master/src/Android/Avalonia.Android/Platform/Storage/AndroidStorageProvider.cs#L58

**Reply by maxkatz6** [reply]
Created issue - https://github.com/AvaloniaUI/Avalonia/issues/14506
If anybody wants to contribute, help is welcomed.

**Reply by maxkatz6** [reply]
Created issue - https://github.com/AvaloniaUI/Avalonia/issues/14506
If anybody wants to contribute, help is welcomed.

**Reply by segln** [reply]
Thank you very much for the answer!
